test/libsolidity/syntaxTests/inlineAssembly/blobhash_pre_cancun.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
+    function f() pure external returns (bytes32 ret) {
+        assembly {
+            ret := blobhash()
+        }
```

Comment:
This is incredibly odd - when EVMDialect is created - more specifically the builtins (`m_functions`), `basefee` is not in there - but then when it check for the presence of `basefee` (for Berlin), it does end up being there, which is why it also throws the 'only supported in London` error.

Reply:
Again, I may be mistaken, since I can't see this mechanism anywhere in the code...

Reply:
```
    TypeError 1006: (74-91): "blobbasefee" is not supported by the VM version.
    TypeError 6679: (187-198): The "blobbasefee" instruction is only available for Cancun-compatible VMs (you are currently compiling for "berlin").
    DeclarationError 8678: (180-200): Variable count for assignment to "ret" does not match number of values (1 vs. 0)
```
Confirmed.

Reply:
Hum...so should we add a test for it anyway?

Reply:
Nice. In that case the test is fine as is.

Reply:
Actually, I'd still add the check even if we cannot test it before `cancun` becomes the default. The current test won't fail when we make it the default and we'll forget then.

Reply:
> Again, I may be mistaken, since I can't see this mechanism anywhere in the code...

I think I figured out why it was done this way: https://github.com/ethereum/solidity/pull/14737#discussion_r1450767916. IMO it should really have a comment explaining the intent (or at least better naming).

In any case, another conclusion is that we could just make this error an assert for now to avoid having to ignore it via `error-codes.py` and only make it an error when we make `cancun` the default (the assert will helpfully fail then).

Reply:
Right. I made it an assertion. See https://github.com/ethereum/solidity/pull/14757/commits/d186d1cb237e1440ffe9f409f2d9ef07ec993ed0